lib/floki.ex

```diff
@@ -289,6 +289,33 @@ defmodule Floki do
     Finder.find(html_tree_as_tuple, selector)
   end
 
+  @doc """
+  Finds the first element in an HTML tree by id.
+
+  Returns nil if no element is found.
+
+  This is useful when there are IDs that contain special characters that
+  are invalid when passed as is as a CSS selector.
+  It is similar to the `getElementById` method in the browser.
+
+  ## Examples
+
+      iex> {:ok, html} = Floki.parse_fragment(~s[<p><span class="hint" id="id?foo_special:chars">hello</span></p>])
+      iex> Floki.find_by_id(html, "id?foo_special:chars")
+      {"span", [{"class", "hint"}, {"id", "id?foo_special:chars"}], ["hello"]}
+      iex> Floki.find_by_id(html, "does-not-exist")
+      nil
+
+  """
+  @spec find_by_id(html_tree() | html_node(), String.t()) :: html_tree
+  def find_by_id(html_tree_as_tuple, id)
+      when is_list(html_tree_as_tuple) or is_html_node(html_tree_as_tuple) do
+    case Finder.find(html_tree_as_tuple, %Floki.Selector{id: id}) do
+      [] -> nil
+      [node | _] -> node
+    end
```

Comment:
What about using `List.first/1` here?

```suggestion
    html_tree_as_tuple
    |> Finder.find(%Floki.Selector{id: id})
    |> List.first()
```